Add unit tests for StringUtils, OverlayIcon and PrimitiveType

shafi-1 opened the pull request:
I've analysed your codebase and noticed that `jadx.tests.functional.StringUtils`, `jadx.gui.utils.OverlayIcon` and `jadx.core.dex.instructions.args.PrimitiveType` are not fully tested.
I've written some tests for the methods in these classes with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important in a subsequent PR.

JohnLBergqvist reviewed (changes requested):
You need to remove the diffblue.yml file from the PR, and i'd probably keep this PR to one of the 3 classes. Then if they like them, submit the remaining 2 as seperate PRs.

Braavos6 commented on jadx-core/src/test/java/jadx/tests/functional/StringUtilsTest.java:
```diff
@@ -50,4 +56,133 @@ public void testCharUnescape() {
+	}
+
+	@Test
+	public void testEscapeWhiteSpaceChar1()
```
Can be refactored further.. same as below
`Assert.assertEquals("\\r", methodUnderTest.invoke(null, '\r'));`

Braavos6 commented on build.gradle:
```diff
@@ -41,6 +41,12 @@ allprojects {
+		testCompile group: 'com.diffblue', name: 'deeptestutils', version: '1.9.0'
+		testCompile group: 'powermock', name: 'powermock', version: '1.7.4'
+		testCompile group: 'org.powermock', name: 'powermock-api-mockito2', version: '1.7.4'
+		testCompile group: 'org.powermock', name: 'powermock-module-junit4', version: '1.7.4'
```
Powermock is not used

paulkennethkent replied:
This is still not resolved, you need to remove powermock from the dependencies as it's not used in the tests

shafi-1 replied:
Yeah forgot to commit updated build.gradle, should be removed now

paulkennethkent replied:
Unless my eyes are deceiving me, it's still there.

Braavos6 commented on jadx-core/src/test/java/jadx/tests/functional/StringUtilsTest.java:
```diff
@@ -50,4 +56,133 @@ public void testCharUnescape() {
+	}
+
+	@Test
+	public void testEscapeWhiteSpaceChar1()
```
Private methods.. these can be covered indirectly from the public method calls instead of using `Reflection`

shafi-1 replied:
Tried covering indirectly but it was not possible.

Braavos6 commented on jadx-core/src/test/java/jadx/tests/functional/StringUtilsTest.java:
```diff
@@ -50,4 +56,133 @@ public void testCharUnescape() {
+
+		final char c = '\r';
+		final Class<?> classUnderTest = Reflector.forName("jadx.core.utils.StringUtils");
+		final Method methodUnderTest =
```
if using reflection change the variable name to something more meaningful

Braavos6 commented:
The testing style used by @shafi-1 is different to the testing style which already exists. Most probably the repo owner will ask you to match this style of testing

paulkennethkent reviewed (approved):
Looks good but fix the gradle file

shafi-1 committed:
Add unit tests for StringUtils, OverlayIcon and PrimitiveType

Fully-qualified classname
jadx.tests.functional.StringUtils
jadx.gui.utils.OverlayIcon
jadx.core.dex.instructions.args.PrimitiveType

These tests were written using Diffblue Cover.
